### ISSUE TITLE
Typescript typings: 

### ISSUE THREAD

**Issue by tomitrescak** [apollographql/react-apollo#810]
### Expected Behavior

I should be able to use React.Components with graphql wrapper.
Example:

```ts
const historyQuery = gql`
  query history($solutionId: String) {
    history(solutionId: $solutionId) {
      solutionId
      delta
    }
  }`;

type Data = {
  history: App.Entities.History[]
};

type Props = {
  solutionId: string;
};

const withHistory = graphql<Data, Props>(historyQuery,
  {
    options: (ownProps) => ({
      variables: {
        solutionId: ownProps.solutionId
      }
    })
  });

export class HistoryView extends React.Component<{}, {}> {
}

export default withHistory(HistoryView);
```

Leads to following error:

```
Argument of type 'typeof HistoryView' is not assignable to parameter of type 'CompositeComponent<DefaultChildProps<Props, Data>>'.
  Type 'typeof HistoryView' is not assignable to type 'StatelessComponent<DefaultChildProps<Props, Data>>'.
    Type 'typeof HistoryView' provides no match for the signature '(props: Props & { data?: QueryProps & Data; mutate?: MutationFunc<Data>; } & { children?: ReactNode; }, context?: any): ReactElement<any>'.
```

Looks like you only accept stateless components, although opposite seems to be the truth.
Not sure what is going on.

### Version
- apollo-client@1.4.2
- typescript@2.4.1

**Comment by morriq** [comment]
same here! It's happend with 1.4.0

**Comment by dfischer** [comment]
Same issue for me with `1.4.3` had to downgrade to `1.4.0`

**Comment by hxuanhung** [comment]
in my case apollo-client@1.4.2, I downgraded TS to 2.3.0

**Comment by hxuanhung** [comment]
@dfischer which version of TS that you are on?

**Comment by hxuanhung** [comment]
I had to downgraded to apollo-client@1.4.2 and TS@2.3.3

**Comment by jbaxleyiii** [comment]
I'm working on a fix that will allow this to work in 1.4.4 🌮
